### PR TITLE
Graph generator

### PR DESCRIPTION
**Pull request by ellem95**
Function to generate graphs representing the number of satellites overhead during RA observation.

### CODE REVIEW

File: tests/event_finder/test_event_finder.py [hunk 1]
```diff
@@ -38,7 +38,7 @@ def test_get_window_list(self):
                                       bandwidth=None
                                   )
                                   )
-        overhead_windows = EventFinderRhodesMill(list_of_satellites=list_of_satellites_with_frequency, reservation=reservation, azimuth_altitude_path=None).get_overhead_windows()
+        overhead_windows = EventFinderRhodesMill(list_of_satellites=list_of_satellites_with_frequency, reservation=reservation, azimuth_altitude_path=None, search_time_start=reservation.time.begin, search_time_end=reservation.time.end).get_overhead_windows()
```
Comment (nikojpapa) [inline tests/event_finder/test_event_finder.py:41]: `search_time_start=reservation.time.begin, search_time_end=reservation.time.end` should be TimeWindow?

File: tests/graph_generator/test_graph_generator.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+    @property
+    def _arbitrary_search_window_start(self) -> datetime:
+        return datetime(year=2023, month=4, day=18, hour=23)
+
+    @property
+    def _arbitrary_search_window_end(self) -> datetime:
+        return datetime(year=2023, month=4, day=19, hour=7)
+
+TestGraphGenerator().test_generate_graph()
```
Comment (nikojpapa) [inline tests/graph_generator/test_graph_generator.py:41]:
```suggestion
```

**Review by nikojpapa — CHANGES_REQUESTED** [review]

File: tests/graph_generator/test_graph_generator.py (hunk 2 above)
Comment (ellem95) [reply tests/graph_generator/test_graph_generator.py]: resolved

File: tests/event_finder/test_event_finder.py (hunk 1 above)
Comment (ellem95) [reply tests/event_finder/test_event_finder.py]: resolved

File: satellite_determination/event_finder/event_finder_rhodesmill/event_finder_rhodesmill.py [hunk 3]
```diff
@@ -121,15 +122,15 @@ def get_overhead_windows_slew(self):
...
                             elif (satellite_azimuth > (point.azimuth.degree + half_beamwidth)) or (satellite_azimuth < (point.azimuth.degree - half_beamwidth)):
                                 if sat_in_view_flag == 1:
-                                    exit_events.append(point.time)
+                                    exit_events.append(point.time.replace(tzinfo=pytz.UTC))
                     starting_interval+=time_delta
             if enter_events != exit_events:
                 exit_events.append(self._reservation.time.end)
             enter_and_exit_pairs = zip(enter_events, exit_events)
-            time_windows = [TimeWindow(begin=begin_event, end=exit_event) for begin_event, exit_event in enter_and_exit_pairs]
+            time_windows = [TimeWindow(begin=begin_event.replace(tzinfo=pytz.UTC), end=exit_event.replace(tzinfo=pytz.UTC)) for begin_event, exit_event in enter_and_exit_pairs]
```
Comment (nikojpapa) [inline satellite_determination/event_finder/event_finder_rhodesmill/event_finder_rhodesmill.py:133]: See https://github.com/NSF-Swift/satellite-overhead/pull/25#discussion_r1182815205

**Review by nikojpapa — CHANGES_REQUESTED** [review]

File: satellite_determination/event_finder/event_finder_rhodesmill/event_finder_rhodesmill.py [hunk 4]
```diff
@@ -121,15 +122,15 @@ def get_overhead_windows_slew(self):
                     satellite_azimuth = min(az.degrees, 360 - az.degrees)
                     if (point.altitude.degree - half_beamwidth) <= satellite_altitude <= (point.altitude.degree + half_beamwidth):
                             if (point.azimuth.degree - half_beamwidth) <= satellite_azimuth <= (point.azimuth.degree + half_beamwidth) and sat_in_view_flag == 0:
-                                enter_events.append(point.time)
+                                enter_events.append(point.time.replace(tzinfo=pytz.UTC))
```
[the remark this answers is not in the document]
Comment (ellem95) [reply satellite_determination/event_finder/event_finder_rhodesmill/event_finder_rhodesmill.py]: done

File: satellite_determination/event_finder/event_finder_rhodesmill/event_finder_rhodesmill.py [hunk 5]
```diff
@@ -121,15 +122,15 @@ def get_overhead_windows_slew(self):
...
                             elif (satellite_azimuth > (point.azimuth.degree + half_beamwidth)) or (satellite_azimuth < (point.azimuth.degree - half_beamwidth)):
                                 if sat_in_view_flag == 1:
-                                    exit_events.append(point.time)
+                                    exit_events.append(point.time.replace(tzinfo=pytz.UTC))
```
[the remark this answers is not in the document]
Comment (ellem95) [reply satellite_determination/event_finder/event_finder_rhodesmill/event_finder_rhodesmill.py]: Done

File: satellite_determination/event_finder/event_finder_rhodesmill/event_finder_rhodesmill.py (hunk 3 above)
Comment (ellem95) [reply satellite_determination/event_finder/event_finder_rhodesmill/event_finder_rhodesmill.py]: Done

File: satellite_determination/event_finder/event_finder_rhodesmill/support/overhead_window_from_events.py [hunk 6]
```diff
@@ -37,22 +35,22 @@ def get(self) -> List[OverheadWindow]:
                                      for event_type in EventTypesRhodesmill)
 
         if (len(self._events) == 1) and (self._events[0].event_type == EventTypesRhodesmill.CULMINATES): #handles sat that is in view for entire reservation
-            time_window = TimeWindow(begin=self._reservation.time.begin, end=self._reservation.time.end)
+            time_window = TimeWindow(begin=self._reservation.time.begin.replace(tzinfo=pytz.UTC), end=self._reservation.time.end.replace(tzinfo=pytz.UTC))
```
[the remark this answers is not in the document]
Comment (ellem95) [reply satellite_determination/event_finder/event_finder_rhodesmill/support/overhead_window_from_events.py]: Fixed all.

**Review by nikojpapa — APPROVED** [review]

**Pull request merged**